### DISCUSSION TITLE
convert: image sequence is required `-composite' @ error/mogrify.c/MogrifyImageList/8163.

### DISCUSSION THREAD

**Discussion by SchuylerVS** [ImageMagick/ImageMagick#4548] [Help] [answered]
Hello, 

Looking help solving this error message: convert: image sequence is required `-composite' @ error/mogrify.c/MogrifyImageList/8163.

Here is the script:

magick convert -size 3600x900 xc:none -composite -fill black -font arial -gravity center -size 3600x900! caption:'WWWWWWWWWWWWWWW WWWWWWWWWWWWWWW' textbg.png

**Comment by fmw42** [comment, marked as answer]
Two input images are required before the -composite.  You do not need the composite nor the second image. In ImageMagick 7, use magick not magick convert.

Try

`magick -size 3600x900 -background none -fill black -font arial -gravity center caption:'WWWWWWWWWWWWWWW WWWWWWWWWWWWWWW' textbg.png`

**Reply by SchuylerVS** [reply]
Thanks for such a quick response!

**Comment by SchuylerVS** [comment]
That works, I also found this works:

magick convert -background transparent -fill black -font arial -gravity center -size 3600x900! caption:'WWWWWWWWWWWWWWW WWWWWWWWWWWWWWW' textbg.png

any reason one is better than the other?

**Reply by SchuylerVS** [reply]
> 

Seems that -convert is not needed?

Also, is there a difference between "-size 3600x900!" and "-size 3600x900"? and their placement?
Lastly, is there a difference between "-background transparent" and "-background none"?

**Reply by fmw42** [reply]
ImageMagick 7 is a complete rewrite of convert and renamed magick. But the other tools are still basically ImageMagick 7 and need to be prefaced with magick.  So convert does not. If you do so, then you go back to using ImageMagick 6 syntax and loose the new features of of ImageMagick 7.

The ! means exactly that size. But when using -size it knows to make the size that you specify. You might need the ! when using -resize, but not -size.

transparent and none are the same thing. Either can be used when specifying either -background or xc:

**Reply by SchuylerVS** [reply]
Ok got it. Thank you for all this great help!

**Comment by SchuylerVS** [comment]
@fmw42 Would you be able to help me turn this code into an endpoint and host on Heroku?

**Reply by fmw42** [reply]
What exactly do you mean by "endpoint and host on Heroku"  Sorry, I know nothing about Heroku. What do you mean by endpoint?

**Reply by SchuylerVS** [reply]
Basically i am looking to turn this into an API call...

**Reply by fmw42** [reply]
What API? ImageMagick has many. Please be as specific as possible when asking questions or for help. Syntax is different on Windows vs Linux/Mac OSX and whether using Imagemgick 6 or 7.

**Reply by SchuylerVS** [reply]
Hmmm... i guess i need to do a bit of research.. this is all a bit over my head. I am just trying to be able to use this code snippet that you helped me with. The requirements would be as follows.

nodejs script:
run server
have endpoint that takes text, text color, font, image1 URL, and image2 URL
endpoint pulls image using URL and generates a text image with the text, text color, and text font and saves that to a folder
run the script on the folder to make the output image
upload the output image somewhere as a URL
the api endpoint returns a json like {image: "https://s3.aws.com/some-generated-composite.png"}

**Reply by fmw42** [reply]
Sorry, that is not my expertise. Perhaps someone else might offer to help.

**Reply by SchuylerVS** [reply]
ok got it. Very much appreciate all your help!
